src/csharp/Grpc.Core.Api/AsyncStreamReaderExtensions.cs

```diff
@@ -46,5 +48,27 @@ public static Task<bool> MoveNext<T>(this IAsyncStreamReader<T> streamReader)
 
             return streamReader.MoveNext(CancellationToken.None);
         }
+
+#if NETSTANDARD2_0
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="streamReader"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async static IAsyncEnumerable<T> ReadAllAsync<T>(this IAsyncStreamReader<T> streamReader, [EnumeratorCancellation]CancellationToken cancellationToken = default)
+        {
+            if (streamReader == null)
+            {
+                throw new System.ArgumentNullException(nameof(streamReader));
+            }
+
+            while (await streamReader.MoveNext(cancellationToken))
+            {
+                yield return streamReader.Current;
```

Comment:
we could avoid using yield  and thus avoid the C# 8.0 requirement?

Reply:
Yes

Reply:
Upgrading the grpc/grpc repo to be able to use C# 8.0 is nontrivial and it would also require depending on preview version of SDK, which is something I'd rather avoid.